tests/test_cli.py

```diff
@@ -0,0 +1,226 @@
+        )
+
+        assert result.exit_code == 1
+
+    def test_represent_command(self, runner, sample_workflow_file):
+        """Test the represent command."""
+
+        result = runner.invoke(app, ["represent", str(sample_workflow_file)])
+
+        assert result.exit_code == 0
+        assert "Representing workflow from" in result.stdout
+        assert "cycles:" in result.stdout  # Should contain workflow structure
+        assert "test_workflow" in result.stdout  # Should contain workflow name
+
+    @pytest.mark.usefixtures("aiida_localhost")
+    def test_run_command(self, runner, sample_workflow_file):
+        """Test the run command."""
+        result = runner.invoke(app, ["run", str(sample_workflow_file)])
+
+        assert result.exit_code == 0
+        assert "▶️ Running workflow" in result.stdout
+        assert "✅ Workflow execution finished" in result.stdout
+
+    @pytest.mark.usefixtures("aiida_localhost")
+    def test_run_execution_failure(self, runner, sample_workflow_file):
+        """Test handling of workflow execution failures."""
+
+        with patch("sirocco.workgraph.AiidaWorkGraph.run") as mock_run:
+            mock_run.side_effect = Exception("Execution failed")
+            result = runner.invoke(app, ["run", str(sample_workflow_file)])
+
+            assert result.exit_code == 1
+            assert "❌ Workflow execution failed during run" in result.stdout
+
+    @pytest.mark.usefixtures("aiida_localhost")
+    def test_submit_command_basic(self, runner, sample_workflow_file):
+        """Test the submit command without waiting."""
+
+        # Mock the entire submit process to avoid all validation
+        with patch("sirocco.cli._prepare_aiida_workgraph") as mock_prepare:
+            mock_aiida_wg = Mock()
+            mock_aiida_wg._core_workflow.name = "minimal_submit_test"  # noqa: SLF001
+
+            # Create a simple mock result that bypasses isinstance checks
+            mock_result = Mock()
+            mock_result.pk = 12345
+            mock_aiida_wg.submit.return_value = mock_result
+            mock_prepare.return_value = mock_aiida_wg
+
+            result = runner.invoke(app, ["submit", str(sample_workflow_file)])
+
+            assert result.exit_code == 0
+            assert "🚀 Submitting workflow" in result.stdout
+
+    @pytest.mark.usefixtures("aiida_localhost")
+    def test_submit_command_with_wait(self, runner, sample_workflow_file):
+        """Test the submit command with wait option."""
+        with patch("sirocco.cli._prepare_aiida_workgraph") as mock_prepare:
+            mock_aiida_wg = Mock()
+            mock_aiida_wg._core_workflow.name = "test_workflow"  # noqa: SLF001
+            mock_result = Mock()
+            mock_result.pk = 12345
+            mock_aiida_wg.submit.return_value = mock_result
+            mock_prepare.return_value = mock_aiida_wg
+
+            result = runner.invoke(app, ["submit", str(sample_workflow_file), "--wait", "--timeout", "120"])
```

Comment:
in aiida v2.7.0 we will remove wait option (and now it is any not working)
```suggestion
            result = runner.invoke(app, ["submit", str(sample_workflow_file),  "--timeout", "120"])
```